contribs/osm/src/main/java/org/matsim/contrib/osm/networkReader/SupersonicOsmNetworkReader.java

```diff
@@ -130,6 +130,9 @@ private void convert() {
             // get the from and to nodes for a sub segment of the current way
             ProcessedOsmNode fromOsmNode = nodes.get(way.getNodeIds().get(i - 1));
             ProcessedOsmNode toOsmNode = nodes.get(way.getNodeIds().get(i));
+            if (fromNodeForSegment == null || fromOsmNode == null || toOsmNode == null) {
+                throw new RuntimeException("Ensure you pass the 'completeWays=yes' argument when executing osmosis commands.");
```

Comment:
Thank you! :) It sounds better. I will fix it.

> Will this error occur only if you have used osmosis before?

In my case, I've seen this error only when I've used osmosis to extract osm data using `--bounding-polygon` option.

> Also, it would be helpful to extract this into its own method, since this part of the code is quite lengthy already

I see. I will fix this too. :+1: